### PR TITLE
Add Kotlin static source analysis checks

### PR DESCRIPTION
**Pull request by psukys**
We're moving to Kotlin, so let's have it as clean as possible.

### CODE REVIEW

**Comment by psukys** [comment]
Integration with Jenkins is only possible through extending a plugin that we're using for Github reports. There's no jenkins plugin for detekt yet.

**Comment by psukys** [comment]
@kordianbruck , @pfent , should be ready for review. I've disabled build-fail on cases where duplicates are found.

File: app/build.gradle [hunk 1]
```diff
@@ -1,9 +1,14 @@
+plugins {
+    id 'io.gitlab.arturbosch.detekt' version "1.0.0.RC6-3"
```
Comment (kordianbruck) [inline app/build.gradle]: Can't this be moved to the detekt.gralde file?

File: detekt-config.yml [hunk 2]
```diff
@@ -21,7 +21,6 @@ test-pattern: # Configure exclusions for test sources
 
 build:
   warningThreshold: 5
-  failThreshold: 10
```
Comment (kordianbruck) [inline detekt-config.yml]: Why was this removed?

Comment (psukys) [reply detekt-config.yml]: On similar pace (as other checks) that violations are only reported but do not fail the build process.

File: app/build.gradle (hunk 1 above)
Comment (psukys) [reply app/build.gradle]: I can check, tho I remember that gradle complained before that `plugins` needs to be topmost instruction
